src/v/cluster/partition_manager.cc

```diff
@@ -350,12 +370,53 @@ ss::future<> partition_manager::shutdown(const model::ntp& ntp) {
           "manager",
           ntp)));
     }
-
+    auto shutdown_state = start_tracking_shutdown(partition);
     // remove partition from ntp & raft tables
     _ntp_table.erase(ntp);
     _raft_table.erase(partition->group());
 
-    return do_shutdown(partition);
+    return do_shutdown(std::move(shutdown_state));
```

Comment:
does this work? there is no move constructor in shutdown_state, so the hook is effectively lost?  (move t to object is no longer in the intrusive list), we should probably define a move constructor with swap_nodes() or something?